Piwo.class.php

```diff
@@ -7,29 +10,55 @@ public static function onParserSetup( &$parser ) {
 		//the "python" magic word with execPy()
 		$parser->setFunctionHook( 'piwo', 'Piwo::execPy', Parser::SFH_OBJECT_ARGS );
 	}
+
+  /**
+   * write the given content to the given filePath
+   */
+  public static function writeStringToFile($filePath,$content) {
+		$f = fopen($filePath, "w");
+		fwrite( $f, $content );
+		fclose( $f );
+		unset($f);
+  }
+
 	// Render the output of {{#python:gram}}.
 	public static function execPy( $parser, $frame, $params ) {
 		//The inputs should contain a gram name and sys.argv
 		//The output should also be wikitext.
 		$page = WikiPage::factory( Title::newFromText( 'Gram:' . $frame->expand( $params[0] ) ) );
 		$name = $frame->expand( $params[0] );
+		$base="/tmp/";
+	  $pyCode= $base . $name . ".py";
+    $pyError= $base . $name . ".error";
 		$content = $page->getContent()->getNativeData() . '';
-		$content = "import sys\nsys.path.append('" . getcwd() . "/extensions/Piwo')" . (((strpos($content, "from mw import") or strpos($content, "import mw")) === false) ? "\nimport mw" : "") . "\nsys.stderr = open('/tmp/" . $name . ".error', 'w')\ndel sys\n" . $content . "\nimport sys\nsys.stderr.close()\ndel sys";
```

Comment:
I see the point of changing this because of the stdout and stderr, but I don't see why you're completely removing it. This should still happen:
```php
		if (strpos($content, "from mw import") || strpos($content, "import mw")) {}
		else $content = "import mw\n" . $content;
		$content = "import sys\nsys.path.append('" . __DIR__ . "')\n" . $content;
```

Reply:
This should at least be optional.